### PR TITLE
Latency hdrhistogram

### PR DESCRIPTION
**Pull request by aleksbykov**
A bunch of commits which add the opportunity to run c-s command with/without docker with hdr histogram logging and 
collectiong histograms for various periods

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fgelcer** [comment]
@asias , FYI
if you can have a look at this draft, it would be appreciated

**Review by asias — COMMENTED** [review]
typo?

s/faile/fail

s/noe/no

File: sdcm/stress_thread.py
Comment (asias) [inline sdcm/stress_thread.py]: Can you explain why you change from 10s to 5s here?

File: docker/env/build_n_push.sh
Comment (asias) [inline docker/env/build_n_push.sh:33]: Unrelated change for hdr?

File: test-cases/performance/perf-regression-latency-250gb-with-nemesis.yaml
Comment (asias) [inline test-cases/performance/perf-regression-latency-250gb-with-nemesis.yaml:2]: Is this a TEST only change so you can speed up the test during development?

File: sdcm/stress_thread.py
Comment (asias) [inline sdcm/stress_thread.py:198]: Maybe %HDR% or %HDR_LOG% is better?

File: sdcm/utils/decorators.py
Comment (asias) [inline sdcm/utils/decorators.py]: It was start - X, end + X, 300

now

it is  start + Y, end +Y,  Z

Is this intentionally?

File: sdcm/results_analyze/__init__.py
Comment (asias) [inline sdcm/results_analyze/__init__.py]: typo in commit message?

s/faile/fail

s/noe/no

**Comment by asias** [comment]
@aleksbykov  Here is the release criteria:

https://docs.google.com/document/d/1i4Scus4OW4DMfVJGyBSN7dJF0Mb1XPRkXkBzsde4_40/edit#heading=h.ql9ps8idpvh


With the this patch set, do we measure both 1) and 2)? Can you share a sample of the data from you report with this patch set?

How about 3: cql drop test?


```
Release Criteria 
Success is defined as:
1) In all 10-minute periods, the 99th percentile of requests is less than 15ms.
2) In all periods spanning an entire topology operation, the 99th percentile of requests is less than 10ms.
The topology operations must succeed.
3) The CQL operation rate must not drop more than 5% in any 10-minute period.

Example:

A test runs from 10:05 to 12:02. There are topology/repair operations in the periods 10:25-10:42, 10:53-11:21, 11:31-12:01. To pass, the following must hold:

In each of the periods [10:05-10:10], [10:10-10:20], [10:20-10:30], …, [11:50-14:00], [14:00-14:02] the 99th percentile of latency must be under 15ms.
In each of the periods [10:25-10:42], [10:53-11:21], [11:31-12:01], the 99th percentile of latency must be under 10ms.
All of the operations have succeeded.
```

**Comment by asias** [comment]
@aleksbykov Can you have a short high level explanation with concrete examples what was collected and processed?

For example, say we have 4 loaders, how many c-s processes was started on each loader, what was collected on each c-s process, how do we merge all the hdr logs from each c-s process, Do we align the timestamp on each c-s process, and so on?

File: test-cases/performance/perf-regression-latency-250gb-with-nemesis.yaml
Comment (roydahan) [reply test-cases/performance/perf-regression-latency-250gb-with-nemesis.yaml:2]: Yes

File: sdcm/cluster_docker.py
Comment (mykaul) [inline sdcm/cluster_docker.py:355]: why do you need to do apt update here?

Comment (fruch) [reply sdcm/cluster_docker.py:355]: Docker images usually clean the apt cache to save space, hence update is needed.

File: sdcm/cluster.py
Comment (fruch) [inline sdcm/cluster.py:4568]: Can you add more information as to why this can't work with newer java ? defect number of the issue.

We should strive to fix it, and not work around it.

I.e. is the c-s from scylla docker image also broken like this ? (Since we want to switch to using that one)

Comment (roydahan) [reply sdcm/cluster.py:4568]: There is/was an issue with our c-s and newer JDK.
This is temporary workaround, I think it should be fixed by now. isn't it @aleksbykov ?

**Comment by roydahan** [comment]
@aleksbykov I reviewed most of it and also the sample report.
Please rebase to resolve conflicts, make sure you resolved all comments from @fruch and we can merge.

**Comment by roydahan** [comment]
Waiting for @aleksbykov verifying that it works with c-s on docker based and then we can merge.

File: sdcm/stress_thread.py
Comment (aleksbykov) [reply sdcm/stress_thread.py]: Set as 10 seconds

File: sdcm/utils/decorators.py
Comment (aleksbykov) [reply sdcm/utils/decorators.py]: fixed

File: sdcm/stress_thread.py
Comment (aleksbykov) [reply sdcm/stress_thread.py]: done

File: sdcm/cluster.py
Comment (aleksbykov) [reply sdcm/cluster.py:4655]: check with the latest scylla docker image. cassandra stress with -log hdrfile=file.hdr is still failed to run
i add new docker image which is use java8, once we witched to cs in docker for performance tests this part could be removed. but it required if cs runs as process on instance

**Comment by roydahan** [comment]
@aleksbykov conflicts.
And I need you to stop changing this PR if you want it to get merge.

Please let me know when you don't have any change planned here and all your testing are done.

**Comment by aleksbykov** [comment]
@roydahan , i resolved confilts and finished with this PR at this stage. So it could be reviewed safely. 
We can run cs + hdr with / without docker and enable hdr cs logging by appropriate flag.

**Comment by aleksbykov** [comment]
@roydahan @asias @fgelcer @fruch could you review ?

**Comment by aleksbykov** [comment]
@roydahan @asias @fgelcer @fruch could you review ?

File: sdcm/cluster.py
Comment (fruch) [inline sdcm/cluster.py:5294]: how this is related to this PR ?

File: sdcm/utils/decorators.py
Comment (fruch) [inline sdcm/utils/decorators.py:23]: why this is in comment ?

File: sdcm/utils/decorators.py
Comment (fruch) [inline sdcm/utils/decorators.py:206]: please remove commented code

File: test-cases/performance/perf-regression-latency-250gb-with-nemesis.yaml
Comment (fruch) [inline test-cases/performance/perf-regression-latency-250gb-with-nemesis.yaml:36]: this was renamed
```suggestion
use_prepared_loaders: true
```

File: configurations/stress_images/cs-java8.yaml
Comment (fruch) [inline configurations/stress_images/cs-java8.yaml:1]: this was renamed

```suggestion
use_prepared_loaders: false
```

File: sdcm/cluster.py
Comment (fruch) [inline sdcm/cluster.py:4759]: this was renamed

```suggestion
            if self.params.get("use_prepared_loaders"):
```

**Comment by fruch** [comment]
@aleksbykov please rebase, and fix conflicts

**Review by fruch — CHANGES_REQUESTED** [review]
few things changed and revert in the cassndra-stress code
also you'll need to rebuild the docker image

File: docker/env/version
Comment (fruch) [inline docker/env/version:1]: next time wait until the final review/merge to take a version, use something like `1.28-hdr-1`

Comment (aleksbykov) [reply docker/env/version:1]: i used the 1.22-hdr, but number run to fast

File: sdcm/cluster.py
Comment (aleksbykov) [reply sdcm/cluster.py:4782]: fixed

File: configurations/stress_images/cs-java8.yaml
Comment (aleksbykov) [reply configurations/stress_images/cs-java8.yaml:1]: fixed

File: test-cases/performance/perf-regression-latency-250gb-with-nemesis.yaml
Comment (aleksbykov) [reply test-cases/performance/perf-regression-latency-250gb-with-nemesis.yaml:36]: fixed

File: sdcm/utils/decorators.py
Comment (aleksbykov) [reply sdcm/utils/decorators.py:23]: There are special methods in tester, so no need to import them directly. removed

File: sdcm/utils/decorators.py
Comment (aleksbykov) [reply sdcm/utils/decorators.py:206]: removed

**Comment by aleksbykov** [comment]
@roydahan @fruch

**Review by fruch — APPROVED** [review]

**Pull request merged**
